core/toolbox.js

```diff
@@ -265,92 +265,99 @@ Blockly.Toolbox.prototype.position = function() {
  * @private
  */
 Blockly.Toolbox.prototype.populate_ = function(newTree) {
-  var rootOut = this.tree_;
-  var that = this;
-  rootOut.removeChildren();  // Delete any existing content.
-  rootOut.blocks = [];
-  var hasColours = false;
-  function syncTrees(treeIn, treeOut, pathToMedia) {
-    var lastElement = null;
-    for (var i = 0, childIn; childIn = treeIn.childNodes[i]; i++) {
-      if (!childIn.tagName) {
-        // Skip over text.
-        continue;
-      }
-      switch (childIn.tagName.toUpperCase()) {
-        case 'CATEGORY':
-          var childOut = rootOut.createNode(childIn.getAttribute('name'));
-          childOut.blocks = [];
-          treeOut.add(childOut);
-          var custom = childIn.getAttribute('custom');
-          if (custom) {
-            // Variables and procedures are special dynamic categories.
-            childOut.blocks = custom;
+  this.tree_.removeChildren();  // Delete any existing content.
+  this.tree_.blocks = [];
+  this.hasColours_ = false;
+  this.syncTrees_(newTree, this.tree_, this.workspace_.options.pathToMedia);
+
+  if (this.tree_.blocks.length) {
+    throw 'Toolbox cannot have both blocks and categories in the root level.';
+  }
+
+  // Fire a resize event since the toolbox may have changed width and height.
+  Blockly.resizeSvgContents(this.workspace_);
+};
+
+
+/**
+ * Sync trees of the toolbox.
+ * @param {Node} treeIn DOM tree of blocks, or null.
+ * @param {Blockly.Toolbox.TreeControl} treeOut
```

Comment:
Now that this isn't anonymous you should be able to get rid of the treeOut parameter and just used this.tree_ in the function.